activesupport/lib/active_support/dependencies.rb

```diff
@@ -421,13 +421,13 @@ def loadable_constants_for_path(path, bases = autoload_paths)
 
       bases.each do |root|
         expanded_root = File.expand_path(root)
-        next unless %r{\A#{Regexp.escape(expanded_root)}(/|\\)} =~ expanded_path
```

Comment:
I assume the `\\` was for Windows support. But then `nesting[0] == ?/` was kinda breaking it. Not sure what to do here, but I can test for both.